Android/MemeShareApp_Antriksh_Batra/app/src/main/java/com/example/memeshare/MainActivity.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.memeshare
+
+import android.content.Intent
+import android.graphics.drawable.Drawable
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import android.widget.ProgressBar
+import android.widget.Toast
+import com.android.volley.Request.Method.GET
+import com.android.volley.Response
+import com.android.volley.VolleyError
+import com.android.volley.toolbox.JsonObjectRequest
+import com.android.volley.toolbox.StringRequest
+import com.android.volley.toolbox.Volley
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import java.lang.reflect.Method
+
+class MainActivity : AppCompatActivity() {
+    var currentImageurl:String?=null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+
+        loadMeme()
+    }
+
+   private fun loadMeme() {
+       val progressBar=findViewById<ProgressBar>(R.id.progressBar)
+       progressBar.visibility=View.VISIBLE
+       // Instantiate the RequestQueue.
+       val url = "https://meme-api.herokuapp.com/gimme"
+
+// Request a string response from the provided URL.
+       val jsonObjectRequest = JsonObjectRequest(GET, url, null,
+               { response ->
+
+                   currentImageurl = response.getString("url")
+                   val memeImageView = findViewById<ImageView>(R.id.memeImageView)
+                   Glide.with(this).load(currentImageurl).listener(object:RequestListener<Drawable>{
+
+                       override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
+                           progressBar.visibility=View.GONE
+                           return false
+                       }
+
+                       override fun onResourceReady(resource: Drawable?, model: Any?, target: Target<Drawable>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
+                           progressBar.visibility=View.GONE
+                           return false
+                       }
+                   }).into(memeImageView)
+               },
+               {
+                   Toast.makeText(this, "Something went wrong", Toast.LENGTH_SHORT).show()
+               })
+       // Add the request to the RequestQueue
+       MySingleton.getInstance(this).addToRequestQueue(jsonObjectRequest)
+   }
+
+
```

Comment:
Consider formatting, don't push in a hurry. IMHO these commits are not going to give you a medal in hacktoberfest.